conans/client/rest/conan_requester.py

```diff
@@ -97,20 +97,14 @@ def add_auth(self, url, kwargs):
 
 
 class ConanRequester:
+    _http_requester_imp = None
```

Comment:
For some reason having these 2 objects look like a bit extra complicated to me, but maybe I am missing something. Why not just something like?:
```python
class ConanRequester:
    if hasattr(requests, "Session"):
            _http_requester = requests.Session()
        else:
            _http_requester = requests

    def __init__(self, config, cache_folder=None):
        if hasattr(requests, "Session"):
            adapter = HTTPAdapter(max_retries=self._get_retries(config))
            self._http_requester.mount("http://", adapter)
            self._http_requester.mount("https://", adapter)
```